CHANGELOG.md

```diff
@@ -1,5 +1,364 @@
+  
+  - **Add HTTP prefix to publish command output. - [elithrar], [pull/198]**
+  
+  Prefix "https://" in front of the "script available" output to allow shells to automatically detect it as a link. Many shells will allow you to click directly on the URL from inside the terminal (such as iTerm's "CMD-Click"), making it much easier to navigate to your subdomains, or any published Workers applications.
+  
+  [elithrar]: https://github.com/elithrar
+  [pull/198]: https://github.com/cloudflare/wrangler/pull/198
+  
+  - **Build: add message and emoji - [xtuc], [pull/193]**
+  
+  The `wrangler` team _really_ loves emoji, so we made sure to send a little bit of ✨ cheer ✨ your way, via a new message and emoji, whenever you use the `build` subcommand. 💌
+  
+  [xtuc]: https://github.com/xtuc
+  [pull/193]: https://github.com/cloudflare/wrangler/pull/193
+
+- ### 🤕 Fixes
+
+  - **Fix issue previewing a project with KV namespaces - [ashleymichal], [pull/353]**
+
+  This PR fixes a critical bug where using `wrangler preview` on a project with [Workers KV][kv] namespaces causes the command to throw an error.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/353]: https://github.com/cloudflare/wrangler/pull/353
+
+  - **Enforce one Webpack entry in configuration - [xtuc], [pull/245]**
+
+  `wrangler` now returns an error during the build process if you use a webpack configuration with more than one export – `wrangler` needs to have a single known export from webpack to know what to build!
+
+  [xtuc]: https://github.com/xtuc
+  [pull/245]: https://github.com/cloudflare/wrangler/pull/245
+
+  - **Update default template for Rust project type - [EverlastingBugstopper], [pull/309]**
+
+  Previously, when passing `--type rust` to `wrangler generate`, the only indication that it worked was that the type in `wrangler.toml` was `rust`. There were no Rust files in the default template, for a Rust-type project. Now, when running `wrangler generate --type rust`, `wrangler` will use [rustwasm-worker-template](https://github.com/cloudflare/rustwasm-worker-template) when generating a new project.
+
+  [pull/309]: https://github.com/cloudflare/wrangler/pull/309
+
+  - **Stop cleaning webpack build artifacts - [EverlastingBugstopper], [pull/307]**
+
+  The configuration for Webpack projects in Wrangler was over-eager to clean build artifacts, which in the worst-case, caused Wrangler to remove source code for developers' projects - oops! This fix relaxes the Webpack cleaning process, to ensure that building a Wrangler project is a bit safer.
+
+  [pull/307]: https://github.com/cloudflare/wrangler/pull/307
+
+  - **Correct binding format - [xtuc], [pull/260]**
+
+  Previously, `wrangler` was incorrectly sending up a `binding` object[] to the Cloudflare API, whenever we attempted to update a script's bindings. This fix renames it to `bindings`, and uses an array, as per the Cloudflare API requirements.
```

Comment:
this object[] is a lil weird.. not actually sure what you meant here

Reply:
i... have no idea

removing!